### PR TITLE
Language Change for issue #38659

### PR DESCRIPTION
**Pull request by coolcat-creations**
Pull Request for Issue #38659 .

### Summary of Changes
Changed the + symbol to "then" in System Panel
Added "then" between the keys in the xml text and made it easier to understand the shortcuts

Before:
J + X  Keyboard Shortcuts

J then A Save
J then S Save & Close
J then Q Cancel 
J then N New
J then F Search 
J then O Options
J then H Help
J then M Toggle 
J then X Overview
J then D Home 

After:
J then X Keyboard Shortcuts

Enables keyboard shortcuts on the administrator site, which can be provided by other plugins and includes directly the following list of shortcuts:

J then A Apply
J then S Save & Close
J then Q Cancel (Quit)
J then N New
J then F Search (Find)
J then O Options
J then H Help
J then M Toggle **M**enu
J then X Overview / E**x**tras
J then D Home **D**ashboard

### Testing Instructions
Go to System and see that the text on the system Panel changed from:
grafik
to 
grafik

Go to System Plugin Shortcuts and see the description changed from:
grafik

to 
grafik

Same should be visible in the modal window when you press J then X

### CODE REVIEW

**Comment by Fedik** [comment]
Dashboard and plugin description works, but In modal without `then`.

**Comment by coolcat-creations** [comment]
> Dashboard and plugin description works, but In modal without `then`.

Yes thats what I wrote :-)

**Comment by Fedik** [comment]
Okay, I thought you will change both :)

**Comment by coolcat-creations** [comment]
I don't know which file to use to change it, no idea why it's not the same as the plugin description :-(

**Comment by chmst** [comment]
Maybe you can make the "then" in smaller font on the Dashboard? 

You changed "Save" to "Apply". It is true that the a button stands for apply, but we have in all toolbars the button SAVE, not Apply.

**Comment by Fedik** [comment]
> no idea why it's not the same as the plugin description 

it is dinamicaly generated here https://github.com/joomla/joomla-cms/blob/febdab61581ffdffb941db3aa7dd368bc083a30e/build/media_source/plg_system_shortcut/js/shortcut.es6.js#L103-L114

**Comment by coolcat-creations** [comment]
> Maybe you can make the "then" in smaller font on the Dashboard?
> 
> You changed "Save" to "Apply". It is true that the a button stands for apply, but we have in all toolbars the button SAVE, not Apply.

Thank you I changed it. Better now?

**Comment by Fedik** [comment]
Yeah, well, not trivial, require new string for `Joomla.Text._('then')` :)

**Comment by coolcat-creations** [comment]
> > no idea why it's not the same as the plugin description
> 
> it is dinamicaly generated here
> 
> https://github.com/joomla/joomla-cms/blob/febdab61581ffdffb941db3aa7dd368bc083a30e/build/media_source/plg_system_shortcut/js/shortcut.es6.js#L103-L114

I Found that file but where has this file the "title" from? Because I want to change them too.

**Comment by Fedik** [comment]
It comes from plugin https://github.com/joomla/joomla-cms/blob/f520c98c64efb538e70e918acffb60d07e4d6b1e/plugins/system/shortcut/src/Extension/Shortcut.php#L119-L139

> why doesn't it take the existing language override?

The modal shows User defined combination, and so the modal content is dinamicaly generated.

**Comment by Fedik** [comment]
> The modal shows User defined combination,

I mean combination that may come from other extennsion with `addShortcuts` event,

**Comment by Fedik** [comment]
But probably, it is fine as you already made,  it is less confusing on dashboard now.

**Comment by coolcat-creations** [comment]
Yes but this would not change the titles as in my xml change

**Comment by brianteeman** [comment]
1. please dont use `<b>` it should not be used for styling text https://developer.mozilla.org/en-US/docs/Web/HTML/Element/b
2. I see what you are doing with highlighting one letter but it only really works in english
3. and you had to invent extras for options which really doesnt make sense.

**Comment by coolcat-creations** [comment]
What does X stand for ? @brianteeman

**Comment by brianteeman** [comment]
it doesnt stand for anything

**Comment by coolcat-creations** [comment]
> it doesnt stand for anything

Let's write the meaning of the shortcuts into brackets so people will remember easier and understand ...

**Comment by brianteeman** [comment]
as stated before - it only makes sense in english

also look at any keyboard shortcut system. there doesnt have to be and often isnt any connection between the key and the action

image

**Comment by dgrammatiko** [comment]
@Fedik should I mention XSS, or the project will also tolerate this one?

**Comment by coolcat-creations** [comment]
> @Fedik should I mention XSS, or the project will also tolerate this one? 

Do you see a security issue in a language change ? 😳

**Comment by dgrammatiko** [comment]
@coolcat-creations not on your side, the existing js code has them. It just need adding the sanitizer in the insertAdjusentHTML call. Again, it’s not about the changes in this pr but the code that was released with 4.2.0

**Comment by coolcat-creations** [comment]
@dgrammatiko maybe it should be reported rather to the JSST ? :-)

**Comment by Fedik** [comment]
@coolcat-creations I think @brianteeman is right, better remove `<b>` on first letter,
There not always will be a case when first letter will refer to "shortcut", especialy for non Eglish translations.

**Comment by coolcat-creations** [comment]
Ok, I applied the requested changes and added also a language string. I cant test the build (es6 file change) myself because I can't get composer and npm running.

**Comment by Fedik** [comment]
Almost :wink: 

image

**Comment by Fedik** [comment]
Add new js string around here:
https://github.com/joomla/joomla-cms/blob/f520c98c64efb538e70e918acffb60d07e4d6b1e/plugins/system/shortcut/src/Extension/Shortcut.php#L94-L97

```
Text::script('PLG_SYSTEM_SHORTCUT_THEN');
```

**Comment by coolcat-creations** [comment]
@Fedik thanks for your help I will add it :-)

**Comment by Fedik** [comment]
About XSS, I will do separated PR, later, after this one will be merged.

**Comment by coolcat-creations** [comment]
Added :-)

**Comment by Fedik** [comment]
Maybe also make it with `<small>`, here how it looks 
Normal
Screenshot 2022-09-02_11-05-31

With `<small>`:
Screenshot 2022-09-02_11-05-05

Just thought, I am fine with any :)

**Comment by Fedik** [comment]
> Added :-)

Works now :)

**Comment by brianteeman** [comment]
was it intentional to split each to a new line?

**Comment by coolcat-creations** [comment]
No, not at all - but how to prevent it? I only changed the + to "then" ?

> was it intentional to split each to a new line?

**Comment by brianteeman** [comment]
the description uses ul lists
the  popup uses dl lists

@chmst can correct me but I think they both should be dl (definition lists)

To put dt and dd on the same line its just css

**Comment by coolcat-creations** [comment]
As this PR is about language change and not semantic, maybe we could merge it first and then move to semantics and the xss and css stuff? One step at the time would increase the speed of improvements :) that would be great!

File: build/media_source/plg_system_shortcut/js/shortcut.es6.js [hunk 1]
```diff
@@ -103,11 +103,11 @@
     let dl = '<dl>';
     dlItems.forEach((titles, shortcut) => {
       dl += '<dt><kbd>J</kbd>';
-      shortcut.split('+').forEach((key) => {
-        dl += ` <kbd>${key.trim()}</kbd>`;
+      shortcut.split('+').forEach(key => {
```
Comment (richard67) [inline build/media_source/plg_system_shortcut/js/shortcut.es6.js]:
```suggestion
      shortcut.split('+').forEach((key) => {
```
Fix JS code style error reported by drone.

File: build/media_source/plg_system_shortcut/js/shortcut.es6.js [hunk 2]
```diff
@@ -103,11 +103,11 @@
...
+        dl += ` ${Joomla.Text._('PLG_SYSTEM_SHORTCUT_THEN')} <kbd>${key.trim()}</kbd>`;
       });
       dl += '</dt>';
-      titles.forEach((title) => {
+      titles.forEach(title => {
```
Comment (richard67) [inline build/media_source/plg_system_shortcut/js/shortcut.es6.js]:
```suggestion
      titles.forEach((title) => {
```
Fix JS code style error reported by drone.

**Comment by coolcat-creations** [comment]
Is there a future for this PR to be merged if tested or it it obligated to change semantics and css? If semantics and css need to be changed maybe someone else can take over and I close this one.

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 0e3bb1fc099af67e51212764f73e60488cf3bacc<br /><br />The aim of this PR was to improve the text, and it is done, so I count it as great success ;)
The styling can be fixed separately, by people who better in it.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38667">issues.joomla.org/tracker/joomla-cms/38667</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38667">issues.joomla.org/tracker/joomla-cms/38667</a>.</sub>

**Comment by HLeithner** [comment]
thanks

**Pull request merged**